src/rrd_daemon.c

```diff
@@ -1495,7 +1495,7 @@ static int handle_request_pending(HANDLER_PROTO) /* {{{ */
   if (ci == NULL)
   {
     pthread_mutex_unlock(&cache_lock);
-    rc = send_response(sock, RESP_ERR, "%s\n", rrd_strerror(ENOENT));
+    rc = send_response(sock, RESP_OK, "updates pending\n");
     goto done;
   }
 
```

Comment:
wouldn't it make more sense to just execute the following for loop only IF ci != NULL instead of duplicating the code above ?